lockup/src/lib.rs

```diff
@@ -222,12 +212,23 @@ impl LockupContract {
             lockup_timestamp: lockup_timestamp.map(|d| d.0),
             transfers_information,
         };
-        let vesting_information = if let Some(vesting_schedule) = vesting_schedule {
-            vesting_schedule.assert_valid();
-            VestingInformation::Vesting(vesting_schedule)
-        } else {
-            VestingInformation::None
+        let vesting_information = match vesting_schedule {
+            None => {
+                assert!(
+                    foundation_account_id.is_none(),
+                    "Foundation account can't be added without vesting schedule"
+                );
+                VestingInformation::None
+            }
+            Some(VestingScheduleOrHash::VestingHash(hash)) => VestingInformation::VestingHash(hash),
+            Some(VestingScheduleOrHash::VestingSchedule(vs)) => {
+                VestingInformation::VestingSchedule(vs)
+            }
         };
+        assert!(
```

Comment:
This check looks irrelevant to adding vesting hash. Did we miss a bug in the past?

Reply:
Not really. Originally, I wrote the vesting schedule to be used for linear release without realizing we need both linear release and vesting schedule. So it was possible to have vesting schedule without foundation account in the past.